agatha/construct/semrep_util.py

```diff
@@ -306,6 +306,40 @@ def _get_flags(self, input_path:Path, output_path:Path)->List[str]:
     res.append(str(output_path))
     return res
 
+  def _run_gracefully(self, input_path:Path, output_path:Path)->None:
+    """
+    Runs semrep safely. If a sentence causes a semrep error, 
+    logs the error and re-tries execution with next input.
+    Overall, encapsulate run functionality and use @320 as a
+    wrapper.
+    Args:
+      input_path: Path to the Semrep input file
+    """
+    if not self.metamap_server.running():
+      self.metamap_server.start()
+
+    input_path = Path(input_path)
+    assert input_path.is_file(), f"Failed to find {input_path}"
+    assert not output_path.exists(), f"Refusing to overwrite {output_path}"
+    cmd = self._get_flags(input_path, output_path)
```

Comment:
If you take a look at `_get_flags` you'll see that the last two parameters are whjat determine the input and output files. This means that the SemRep process will not be expecting stdin or stdout input. You'll need to make a change there as well.

-